subsys/net/ip/tcp2.c

```diff
@@ -689,16 +691,15 @@ static int tcp_header_add(struct tcp *conn, struct net_pkt *pkt, uint8_t flags,
 
 	memset(th, 0, sizeof(struct tcphdr));
 
-	th->th_sport = conn->src.sin.sin_port;
-	th->th_dport = conn->dst.sin.sin_port;
-
-	th->th_off = 5;
-	th->th_flags = flags;
-	th->th_win = htons(conn->recv_win);
-	th->th_seq = htonl(seq);
+	UNALIGNED_PUT(conn->src.sin.sin_port, &th->th_sport);
+	UNALIGNED_PUT(conn->dst.sin.sin_port, &th->th_dport);
+	th_off(th) = 5;
```

Comment:
Not sure about that line. I assume `th_off()` getter is there to be in sync with other getters of `tcphdr`. However here this getter is used as it would return a reference, not a value. This would be wrong with all other getters.